### ISSUE TITLE
[Dependency-Injection] Upgrade to 5.2.4 creates unneeded dependency for PropertyInfo in test environment

### ISSUE THREAD

**Issue by ovgray** [symfony/symfony#40395]
**Symfony version(s) affected**: 5.2.4

**Description**
When processing a submitted form in test env a controller throws an exception:

> LogicException:
> Unable to use the "Symfony\Component\PropertyInfo\Extractor\PhpDocExtractor" class as the "phpdocumentor/reflection- docblock" package is not installed.
> 
This did not occur with the same code in sf 5.2.3. It does not occur when the same code runs in dev mode.

If I revert symfony/dependency-injection to 5.2.3, the controller does not throw the exception in test env.

Here is the stack trace back to the request handling:

> LogicException:
> Unable to use the "Symfony\Component\PropertyInfo\Extractor\PhpDocExtractor" class as the "phpdocumentor/reflection-docblock" package is not installed.
> 
>   at  [url deleted]\vendor\symfony\property-info\Extractor\PhpDocExtractor.php:63
>   at Symfony\Component\PropertyInfo\Extractor\PhpDocExtractor->__construct()
>      ( [url deleted]\var\cache\test\ContainerWPiJKPq\App_KernelTestDebugContainer.php:2606)
>   at ContainerWPiJKPq\App_KernelTestDebugContainer->ContainerWPiJKPq\{closure}()
>      ( [url deleted]\vendor\symfony\property-info\PropertyInfoExtractor.php:108)
>   at Symfony\Component\PropertyInfo\PropertyInfoExtractor->extract(object(RewindableGenerator), 'getTypes', array('Symfony\\Component\\Form\\Form', 'parent', array()))
>      ( [url deleted]\vendor\symfony\property-info\PropertyInfoExtractor.php:74)
>   at Symfony\Component\PropertyInfo\PropertyInfoExtractor->getTypes('Symfony\\Component\\Form\\Form', 'parent')
>      ( [url deleted]\vendor\symfony\validator\Mapping\Loader\PropertyInfoLoader.php:68)
>   at Symfony\Component\Validator\Mapping\Loader\PropertyInfoLoader->loadClassMetadata(object(ClassMetadata))
>      ( [url deleted]\vendor\symfony\validator\Mapping\Loader\LoaderChain.php:54)
>   at Symfony\Component\Validator\Mapping\Loader\LoaderChain->loadClassMetadata(object(ClassMetadata))
>      ( [url deleted]\vendor\symfony\validator\Mapping\Factory\LazyLoadingMetadataFactory.php:101)
>   at Symfony\Component\Validator\Mapping\Factory\LazyLoadingMetadataFactory->getMetadataFor(object(Form))
>      ( [url deleted]\vendor\symfony\validator\Validator\RecursiveContextualValidator.php:306)
>   at Symfony\Component\Validator\Validator\RecursiveContextualValidator->validateObject(object(Form), '', array('Default'), 1, object(ExecutionContext))
>      ( [url deleted]\vendor\symfony\validator\Validator\RecursiveContextualValidator.php:138)
>   at Symfony\Component\Validator\Validator\RecursiveContextualValidator->validate(object(Form), null, array('Default'))
>      ( [url deleted]\vendor\symfony\validator\Validator\RecursiveValidator.php:93)
>   at Symfony\Component\Validator\Validator\RecursiveValidator->validate(object(Form), null, null)
>      ( [url deleted]\vendor\symfony\validator\Validator\TraceableValidator.php:66)
>   at Symfony\Component\Validator\Validator\TraceableValidator->validate(object(Form))
>      ( [url deleted]\vendor\symfony\form\Extension\Validator\EventListener\ValidationListener.php:50)
>   at Symfony\Component\Form\Extension\Validator\EventListener\ValidationListener->validateForm(object(PostSubmitEvent), 'form.post_submit', object(EventDispatcher))
>      ( [url deleted]\vendor\symfony\event-dispatcher\EventDispatcher.php:230)
>   at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(Closure), object(Closure)), 'form.post_submit', object(PostSubmitEvent))
>      ( [url deleted]\vendor\symfony\event-dispatcher\EventDispatcher.php:59)
>   at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(PostSubmitEvent), 'form.post_submit')
>      ( [url deleted]\vendor\symfony\event-dispatcher\ImmutableEventDispatcher.php:33)
>   at Symfony\Component\EventDispatcher\ImmutableEventDispatcher->dispatch(object(PostSubmitEvent), 'form.post_submit')
>      ( [url deleted]\vendor\symfony\form\Form.php:668)
>   at Symfony\Component\Form\Form->submit(array(), true)
>      ( [url deleted]\vendor\symfony\form\Extension\HttpFoundation\HttpFoundationRequestHandler.php:109)
>   at Symfony\Component\Form\Extension\HttpFoundation\HttpFoundationRequestHandler->handleRequest(object(Form), object(Request))
>      ( [url deleted]\vendor\symfony\form\Form.php:490)
>   at Symfony\Component\Form\Form->handleRequest(object(Request))
>      ( [url deleted]\src\Controller\UpdateController.php:255)

**How to reproduce**
My skills are not up to figuring out why dependency-injection creates a TestDebugContainer that adds PhpDocExtractor to  PropertyInfoExtractor when evidently it does not do that when creating a DevDebugContainer.

**Comment by xabbuh** [comment]
Can you create a small example application that allows to reproduce your issue?

**Comment by ovgray** [comment]
In the course of trying to create an example application, I first looked around to see what it was that would only be included in the container in test mode that might have made DependencyInjection think "phpdocumentor/reflection-docblock" was available for inclusion in PhpDocExtractor when it actually was not.

I had a few Behat Context classes (added to container only in test env) that included bin/.phpunit/phpunit-8.5-0/vendor/autoload.php in order to use phpunit assertions. It turned out that if I commented the inclusions out, there was no longer a problem

I don't know why those inclusions only became a problem when DependencyInjection updated from 5.2.3 to 5.2.4, but I will not suggest that this needs fixing symfony.  I rewrote the classes to use webmozart assertions instead. That solved the problem.

I will close this issue.
